### ISSUE TITLE
Accessibility Issue - Switch Control Missing Associated Label

### ISSUE THREAD

**Issue by parul-26** [carbon-design-system/carbon#17525]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.40.0

### React version

18.2.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description
the error we are getting :
<img width="1726" alt="image" src="https://github.com/user-attachments/assets/ec84d411-d4fd-41b6-b8e1-7407f1b0e0cc">

In the Carbon Design System, the toggle control with the switch role is missing an associated label. This causes a  accessibility violation, as it makes the control unusable for screen reader users.
![image](https://github.com/user-attachments/assets/bbcb77c8-020c-4572-8b06-b12abe563c41)

### Reproduction/example

Component: Toggle (With Accessible Labels)  URL:https://react.carbondesignsystem.com/?path=/story/components-toggle--with-accessible-labels

### Steps to reproduce

1. Open the Carbon Design System documentation for the Toggle component.
2. Navigate to the "With Accessible Labels" section.
3. Inspect the toggle control with the switch role.
4. Observe that the control lacks an associated label, resulting in an accessibility violation.

### Suggested Severity

None

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by philljenkins** [comment]
In the screenshot with this issue, either the 
- "**_Alerts and events_**" should be the programmatically associated and visible label for the switch, or the
- "**_If you disabled the card, it will not be displayed on the home page_**" as the label.

and open a separate new Carbon issue to fix No. 3 switch example:
- [ ] Fix label in No. 3 "**_Internal aria-label toggle_**" replacing `aria-label` with `aria-labelledby` as explained above.
